osu.Game/Online/API/Requests/Responses/APIBeatmapSet.cs

```diff
@@ -111,6 +111,12 @@ public string AuthorString
         [JsonProperty(@"language")]
         public BeatmapSetOnlineLanguage Language { get; set; }
 
+        [JsonProperty(@"current_nominations")]
+        public BeatmapSetOnlineNominations[] CurrentNominations { get; set; } = Array.Empty<BeatmapSetOnlineNominations>();
+
+        [JsonProperty(@"related_users")]
+        public APIUser[] RelatedUsers { get; set; } = Array.Empty<APIUser>();
```

Comment:
These should be marked nullable and have a default of `null`, because as far as I can tell from the web-side implementations, these properties are contextual includes that need not be present in every single game-side context that uses `APIBeatmapSet`. And initialising with `Array.Empty<>()` is potentially misleading consumers that there is valid data to consume there, while there is none.

This is leaving aside the fact that this class is similarly bloated as `APIUser` is, and should really be multiple context-sensitive classes, but the ship has likely long sailed on this too.